libraries/AP_Scripting/applets/arming-checks.lua

```diff
@@ -0,0 +1,620 @@
+    -- RTL_ALTITUDE will be correct for either plane or copter
+    if (RTL_ALTITUDE ~= nil and (RTL_ALTITUDE:get() or 300) > alt_legal_max) then
+        return false
+    end
+    return true
+end
+
+-- is the Q RTL altitude (quadplane) legal? This defaults to 120m (400') as set in ARM_ALT_LEGAL
+local function q_rtl_alt_legal()
+    -- Plane specific Parameters
+    if (Q_ENABLE:get() or 0) ~= 1 then
+        return true
+    end
+    -- QuadPlane specific Paraemters
+    local q_rtl_alt = Parameter("Q_RTL_ALT"):get() or 0
+    if q_rtl_alt > alt_legal_max then
+        return false
+    end
+    return true
+end
+
+-- is the RTL climb minimum legal? This defaults to 120m (400')
+local function rtl_climb_legal()
+    if (RTL_CLIMB_MIN ~= nil and RTL_CLIMB_MIN:get() > alt_legal_max) then
+        return false
+    end
+    return true
+end
+
+-- Display a message as to whether failsafe will QLand or QRTL
+local function qland_warning()
+    if (Q_ENABLE:get() or 0) ~= 1 then
+        return true
+    end
+    -- Only applies to QuadPlane
+    local q_options = Parameter("Q_OPTIONS"):get() or 0
+    local q_land_option = ((q_options & (1 << 5)) > 0)
+    local q_RTL_option = ((q_options & (1 << 20)) > 0)
+
+    if (not q_land_option) and (not q_RTL_option) then
+        return false
+    end
+    return true -- always ok if not a QuadPlane
+end
+local function qrtl_warning()
+    if (Q_ENABLE:get() or 0) ~= 1 then
+        return true
+    end
+    local q_options = Parameter("Q_OPTIONS"):get() or 0
+    local q_land_option = ((q_options & (1 << 5)) > 0)
+    local q_RTL_option = ((q_options & (1 << 20)) > 0)
+    if q_land_option or q_RTL_option then
+        return false
+    end
+    return true -- always ok if not a QuadPlane
+end
+
+-- A single check to ensure AIRSPEED_STALL < AIRSPEED_MIN < AIRSPEED_CRUISE < AIRSPEED_MAX
+local function airspeed_check()
+    local airspeed_stall = AIRSPEED_STALL:get() or -1
+    local airspeed_min = AIRSPEED_MIN:get() or -1
+    local airspeed_cruise = AIRSPEED_CRUISE:get() or -1
+    local airspeed_max = AIRSPEED_MAX:get() or -1
+
+    if airspeed_stall > 0 then
+        if airspeed_min < airspeed_stall then
+            return false
+        end
+    end
+    if airspeed_cruise < airspeed_min then
+        return false
+    end
+    if airspeed_max < airspeed_cruise then
+        return false
+    end
+    return true
+end
+
+-- If AIRSPEED_STALL is set AIRSPEED_MIN should be at least 25% higher than AIRSPEED_STALL
+local function stallspeed_check()
+    local airspeed_stall = AIRSPEED_STALL:get() or -1
+    local airspeed_min = AIRSPEED_MIN:get() or -1
+
+    if airspeed_stall > 0 then
+        if airspeed_min > (airspeed_stall * 1.25) then
+            return false
+        end
+    end
+
+    return true
+end
+
+-- From the wiki: This is the center of the speed scaling range and should be set close to the normal cruising speed of the vehicle.
+local function scaling_speed_check()
```

Comment:
I'm not a fan of this one. The wiki is correct, however if you change the scaling speed you have to re-tune. We don't want people to start using this script, get the arming check and then just change it in the field without knowing they have to retune.